src/eei.cpp

```diff
@@ -794,6 +803,9 @@ inline int64_t maxCallGas(int64_t gas) {
       // NOTE: this must be declared outside the condition to ensure the memory doesn't go out of scope
       vector<uint8_t> contract_code;
       if (length) {
+        // Perform this check here, rather than inside loadMemory, before we attempt to resize the vector
+        ensureCondition((dataOffset + length) >= dataOffset, InvalidMemoryAccess, "Out of bounds (source) memory copy.");
+        ensureCondition(memory.size() >= (dataOffset + length), InvalidMemoryAccess, "Out of bounds (source) memory copy.");
```

Comment:
Any way we can pull these into helpers? i'm not a fan of code duplication. (We let it pass with the gas calculation with the note that it will be refactored, but it wasn't.)

I suggest a helper like
```
ensureSourceBuffer(uint32_t offset, uint32_t length) {
    ensureCondition((offset + length) >= offset, InvalidMemoryAccess, "Out of bounds (source) memory copy.");
    ensureCondition(memory.size() >= (offset + length), InvalidMemoryAccess, "Out of bounds (source) memory copy.");
}
```